src/utils/getRecentCommitsUtility.js

```diff
@@ -0,0 +1,193 @@
+                        return true;
+                    });
+
+                    // Save each repo's commits data to the database
+                    let currTime = (new Date).getTime();
+                    console.log(`in fetchRecentMonthCommits, storing recent-${repo}-month-commits`);
+                    model_utils.setItem(`recent-${repo}-month-commits`, result);
+                    model_utils.setItem(`recent-${repo}-month-expiry`, currTime);
+                }
+                return result;
+            });
+}
+
+// Fetches recent month commits for top 10 repositories
+function fetchAllRecentMonthCommits(repos, queryTime) {
+    let results = [];
+    let commitersSet = new Set([]);
+    let timeToday = (new Date).getTime();
+
+    // We take only 10 repos just for API quota reasons
+    let splicedRepos = repos.splice(0,10);
+
+    let promises = splicedRepos.map(function mapToEachRepo(repo, i) {
+        return fetch(`https://api.github.com/repos/publiclab/${repo}/commits?since=${queryTime}`)
+                .then(function gotResponse(response) {
+                    if(response.status=="200") {
+                        return response.json();
+                    }
+                })
+                .then(function gotResponseJson(response) {
+                    if(response!=null) {
+                        let partialResult = [];
+                        response.map(function mappingToCommits(commit, i) {
+                            if(commit!=null) {
+                                if(!commitersSet.has(commit.author.login)) {
+                                    commitersSet.add(commit.author.login);
+                                    partialResult.push(commit);
+                                    results.push(commit);
+                                }
+                            }
+                            return true;
+                        });
+                        // Save each repo's commits data to the database
+                        let currTime = (new Date).getTime();
+                        model_utils.setItem(`recent-${repo}-month-commits`, partialResult);
+                        model_utils.setItem(`recent-${repo}-month-expiry`, currTime);
+                        
+                        // Push every repo's contribs to results array
+                        // results.push(partialResult);
+                    }
+                    return true;
+                });
+        });
+
+    return Promise.all(promises)
+           .then(function promisesResolved() {
+                // Store recentMonthCommits and recentMonthCommitsExpiry in the database
+                console.log(`in fetchALLRecentMonthCommits, storing recent-ALL-month-commits`);
+                model_utils.setItem('recent-all-month-commits', results);
+                model_utils.setItem('recent-all-month-expiry', timeToday);
+                return results;
+            });
+}
+
+
+function getCommitsLastWeek(org, repo) {
```

Comment:
Basically it's OK to have one function per file if the functions are this big and each serves a single purpose.

Reply:
Yeah, I will try to refactor this chunk in a follow-up PR...